ssh2/sftp_handle.pyx

```diff
@@ -268,9 +268,15 @@ cdef class SFTPHandle:
         :rtype: int"""
         cdef size_t _size = len(buf)
         cdef char *cbuf = buf
-        cdef ssize_t rc
+        cdef ssize_t rc = 0
         with nogil:
-            rc = c_sftp.libssh2_sftp_write(self._handle, cbuf, _size)
+            while _size > 0:
+                rc = c_sftp.libssh2_sftp_write(self._handle, cbuf, _size)
+                if rc < 0:
```

Comment:
This is not needed anymore.

Reply:
Could you elaborate on this? why not?
Would think that you would keep writing the content of 'cbuf' until it is empty

Reply:
Was referring to `if rc < 0` check. Since the loop was changed back to `> 0`, rc can never be `< 0`.

Reply:
My mistake actually :)